### PR TITLE
Meta: Move UCD/CLDR/TZDB downloaded artifacts to Build/caches

### PR DESCRIPTION
**Pull request by trflynn89**
Partially addresses #16200. I haven't touched PCI or USB ID files in this patch, as those are basically just downloaded raw into their on-disk location.

### CODE REVIEW

File: Meta/Lagom/CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,8 @@ get_filename_component(
 )
 set(SerenityOS_SOURCE_DIR "${SERENITY_PROJECT_ROOT}" CACHE STRING "")
 
+set(SERENITY_CACHE_DIR "${PROJECT_BINARY_DIR}" CACHE STRING "")
```
Comment (ADKaster) [inline Meta/Lagom/CMakeLists.txt:28]: These look like the correct changes. I think it's missing one thing still, which is the interaction with `Meta/serenity.sh test lagom` when there's not already a Lagom build dir created by a super build. 

Not sure how many people are running into that edge case though.

Comment (trflynn89) [reply Meta/Lagom/CMakeLists.txt:28]: Bah, I had a serenity.sh change to cover that case, would've helped if I commited it :expressionless:

**Review by ADKaster — APPROVED** [review]
